esp/src/eclwatch/DynamicESDLQueryWidget.js

```diff
@@ -172,27 +172,28 @@ define([
                 }
             }, domID);
 
-            retVal.on(".dgrid-row-url:click", function (evt) {
-                if (context._onRowDblClick) {
-                    var item = retVal.row(evt).data;
-                    context._onRowDblClick(item);
-                }
-            });
+            ESPUtil.goToPageUserPreference(retVal, "DynamicESDLQueryWidget_GridRowsPerPage").then(function(){
+                retVal.on(".dgrid-row-url:click", function (evt) {
+                    if (context._onRowDblClick) {
+                        var item = retVal.row(evt).data;
+                        context._onRowDblClick(item);
+                    }
```

Comment:
There is no need to put the "on" calls in the "then" - its that startup thats important.